ql/src/java/org/apache/hadoop/hive/ql/metadata/PartitionTree.java

```diff
@@ -121,8 +122,8 @@ List<Partition> addPartitions(List<Partition> partitions, boolean ifNotExists)
    * {@link MetaStoreUtils#getPvals(List, Map)}
    */
   List<Partition> getPartitionsByPartitionVals(List<String> partialPartVals) throws MetaException {
-    if (partialPartVals == null || partialPartVals.isEmpty()) {
-      throw new MetaException("Partition partial vals cannot be null or empty");
+    if (MetaStoreUtils.arePartValsEmpty(partialPartVals)) {
```

Comment:
same as above

Reply:
Add null check in `SessionHiveMetaStoreClient`, here we can follow the server side behavior to support empty values.